### PR TITLE
Authenticate with JSTOR using the site code passed by LMS

### PR DESCRIPTION
**Pull request by jon-betts**
For:

 * https://github.com/hypothesis/via/issues/734
 
Requires:

 * https://github.com/hypothesis/lms/pull/3912

## Testing notes

It would be ideal to test this with a document which isn't publicly accessible, but I need to find one first.

This assumes you have the setup running from https://github.com/hypothesis/lms/pull/3912

 * Visit your JSTOR assigment with the F12 dev tools open
 * Inspect the data being pass to `proxied.pdf` - you should see `site_code=` in the query string
 * It should not be rejected by JSTOR

I've tried this with a restricted document (https://www.jstor.org/stable/j.ctt7zwbts) but we're getting upstream 500's?

### CODE REVIEW

File: tests/unit/via/services/jstor_test.py [hunk 1]
```diff
@@ -26,15 +29,21 @@ def test_is_jstor_url(self, url, is_jstor):
     @pytest.mark.parametrize(
         "url,expected",
         [
-            ("jstor://ARTICLE_ID", "http://jstor.example.com/10.2307/ARTICLE_ID?ip=IP"),
-            ("jstor://PREFIX/SUFFIX", "http://jstor.example.com/PREFIX/SUFFIX?ip=IP"),
+            ("jstor://ARTICLE_ID", "http://jstor.example.com/pdf/10.2307/ARTICLE_ID"),
+            ("jstor://PREFIX/SUFFIX", "http://jstor.example.com/pdf/PREFIX/SUFFIX"),
```
Comment (jon-betts) [inline tests/unit/via/services/jstor_test.py:33]: We don't roll the specific end-point into the ENV var any more

File: README.md [hunk 2]
```diff
@@ -68,23 +68,25 @@ Configuration
...
+| `VIA_HTML_URL`             | The URL of the Via HTML instance to redirect to for proxying HTML pages                                                                                                                                                                                             | `https://viahtml.hypothes.is/proxy` |
+| `VIA_SECRET`               | The secret that must be used to sign URLs to Via's Python endpoints if `SIGNED_URLS_REQUIRED` is on                                                                                                                                                                 |                                     |
+| `JSTOR_API_URL`            | URL for integrating with JSTOR                                                                                                                                                                                                                                      | `http://example.com/api`            |
```
Comment (marcospri) [inline README.md]: `JSTOR_API_URL` is duplicated further up

File: requirements/requirements.in [hunk 3]
```diff
@@ -9,6 +9,7 @@ pyramid-exclog
 pyramid-jinja2
 pyramid-sanity
 pyramid-services
+python-jose[crypto]
```
Comment (marcospri) [inline requirements/requirements.in:12]: Missed opportunity to call this python-pepe (sorry, Spanish joke)

**Review by marcospri — APPROVED** [review]
Looks good, it does what it suppose to do but it's hard to tell if the JSTOR side will behave accordingly.

**Pull request merged**